provider/labels.go

```diff
@@ -78,3 +99,23 @@ func fixLabelNames(prov *tfbridge.ProviderInfo) {
 	}
 
 }
+
+// pulumiLabels represents a field added upstream via a custom diff, as terraform_labels.
+// There is a bug: f this field is nil in the Pulumi state, it will never be added to the state,
+// creating a permanent diff.
+// See also: https://github.com/pulumi/pulumi-gcp/issues/1314
+func ensureLabelPathsExist(paths []resource.PropertyPath) tfbridge.PropertyTransform {
+
+	return func(ctx context.Context, prop resource.PropertyMap) (resource.PropertyMap, error) {
+		obj := resource.NewObjectProperty(prop)
+		for _, path := range paths {
+			// If a `pulumiLabels` field is not set at the expected path, we will set it to an empty property Map,
+			// so Diff can detect it.
+			if _, found := path.Get(obj); !found {
+				path.Set(obj, resource.NewObjectProperty(resource.PropertyMap{}))
```

Comment:
I wasn't sure if `Get` and `Set` would work with globs. They don't. This is fine, since there are no `pulumiLabels` through `lists` or `maps`. I'll add a suggestion for how we can assert that at `make tfgen` time.